python_modules/dagster/dagster/_core/definitions/automation_tick_evaluation_context.py

```diff
@@ -259,16 +244,84 @@ def build_run_requests_from_asset_partitions(
     )
 
 
+def _build_backfill_requests(
+    entity_subsets: Sequence[EntitySubset[EntityKey]],
+    asset_graph: BaseAssetGraph,
+    run_tags: Optional[Mapping[str, str]],
+) -> Tuple[Sequence[RunRequest], Sequence[EntitySubset[EntityKey]]]:
+    """Determines a set of entity subsets that can be executed using a backfill.
+    If any entity subset has size greater than 1, then it and all assets connected
+    to it will be grouped into a backfill. Returns the corresponding backfill
+    run requests, and all entity subsets not handled in this process.
+    """
+    entity_subsets_by_key = {es.key: es for es in entity_subsets}
+    visited: Set[EntityKey] = set()
+    run_requests: List[RunRequest] = []
+
+    def _flood_fill_asset_subsets(k: EntityKey) -> Sequence[EntitySubset[AssetKey]]:
+        if k in visited or k not in entity_subsets_by_key:
+            return []
+        visited.add(k)
+        if isinstance(k, AssetKey):
+            node = asset_graph.get(k)
+            subset = cast(EntitySubset[AssetKey], entity_subsets_by_key.pop(k))
+            return list(
+                itertools.chain(
+                    [subset],
+                    # include all parents and children
+                    *(_flood_fill_asset_subsets(k) for k in node.parent_keys),
+                    *(_flood_fill_asset_subsets(k) for k in node.child_keys),
+                    # include all assets / checks within the execution set
+                    *(
+                        _flood_fill_asset_subsets(k)
+                        for k in asset_graph.get_execution_set_asset_and_check_keys(k)
+                    ),
+                )
+            )
+        else:
+            # if we get an asset check in this code path, it must have been part of
```

Comment:
i'm a little confused by this comment - is this specifically for checks that are run within the asset function (like the `inside` checks you have in the test)? or does it also include asset checks that are defined via `@asset_check` (like the `outside` checks in the test).

Reply:
Yeah the language here isn't the greatest -- the "required execution set" refers to the set of entity keys that must be executed as a unit in order to create a valid run.

So for multi-assets that are not subsettable, all asset keys within that multi-asset will be part of the "get_execution_set_asset_and_check_keys", as will any checks that must run alongside a given asset.

However, non-required asset checks (the "outside" ones) will not be included.